Source/WebCore/rendering/MarkedText.cpp

```diff
@@ -323,12 +322,22 @@ Vector<MarkedText> MarkedText::collectForDocumentMarkers(const RenderText& rende
     return markedTexts;
 }
 
-Vector<MarkedText> MarkedText::collectForDraggedContent(const RenderText& renderer, const TextBoxSelectableRange& selectableRange)
+Vector<MarkedText> MarkedText::collectForDraggedAndTransparentType(const DocumentMarker::Type type, const RenderText& renderer, const TextBoxSelectableRange& selectableRange)
 {
-    auto draggedContentRanges = renderer.draggedContentRangesBetweenOffsets(selectableRange.start, selectableRange.start + selectableRange.length);
+    Type markerType = MarkedText::Type::Unmarked;
+    if (type == DocumentMarker::Type::DraggedContent)
+        markerType = MarkedText::Type::DraggedContent;
+    else if (type == DocumentMarker::Type::TransparentContent)
+        markerType = MarkedText::Type::TransparentContent;
+    else {
+        ASSERT_NOT_REACHED();
+        return { };
+    }
```

Comment:
I think this should be refactored into a lambda which returns an optional `Type` and uses a switch-case